File: Foundation/src/main/kotlin/net/rickiekarp/foundation/config/redis/RedisCacheConfig.java
package net.rickiekarp.foundation.config.redis;

import org.springframework.context.annotation.Bean;
import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;

public class RedisCacheConfig {

    @Bean
    JedisConnectionFactory jedisConnectionFactory() {
        JedisConnectionFactory jedisConFactory = new JedisConnectionFactory();
        jedisConFactory.setHostName("localhost");
        jedisConFactory.setPort(6379);
        return jedisConFactory;
    }

}

File: projects/homebackend/Foundation/src/main/kotlin/net/rickiekarp/foundation/config/redis/RedisCacheConfig.kt
package net.rickiekarp.foundation.config.redis

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.core.env.Environment
import org.springframework.data.redis.connection.RedisStandaloneConfiguration
import org.springframework.data.redis.connection.jedis.JedisConnectionFactory

class RedisCacheConfig {

    @Autowired
    private val env: Environment? = null

    @Bean
    internal fun jedisConnectionFactory(): JedisConnectionFactory {
        val config = RedisStandaloneConfiguration()
        config.hostName = env!!.getProperty("redis.url")!!
        config.port = env.getProperty("redis.port")!!.toInt()
        return JedisConnectionFactory(config)
    }
}
